src/ucp/api/ucp.h

```diff
@@ -4373,6 +4375,55 @@ ucs_status_ptr_t ucp_worker_flush_nbx(ucp_worker_h worker,
                                       const ucp_request_param_t *param);
 
 
+/**
+ * @ingroup UCP_ENDPOINT
+ * @brief UCP endpoint attributes field mask.
+ *
+ * The enumeration allows specifying which fields in @ref ucp_ep_attr_t are
+ * present. It is used to enable backward compatibility support.
+ */
+enum ucp_ep_attr_field {
+    UCP_EP_ATTR_FIELD_NAME = UCS_BIT(0) /**< UCP enpoint name */
+};
+
+
+/**
+ * @ingroup UCP_ENDPOINT
+ * @brief UCP endpoint attributes.
+ *
+ * The structure defines the attributes which characterize the particular
+ * endpoint.
+ */
+typedef struct ucp_ep_attr {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref ucp_ep_attr_field.
+     * Fields not specified in this mask will be ignored.
+     * Provides ABI compatibility with respect to adding new fields.
+     */
+    uint64_t field_mask;
+
+    /**
+     * Tracing and analysis tools can identify the endpoint using this name.
```

Comment:
Endpoint name. Tracing and analysis tools can identify the endpoint using this name.